### PR TITLE
Symbol::hash: Speed up hashing by using SymbolRef order.

### PR DESCRIPTION
**Pull request by jvilk-stripe**
SymbolRef IDs are stable during hashing.

Delurks use of membersStableOrderSlow, which uses alphabetical order which is _much_ more expensive to compute!

Speeds up hashing by ~14%.

### Motivation

Hashing spends ~37% of its time in membersStableOrderSlow.

### Test plan

Covered by existing tests.

### CODE REVIEW

**Comment by jvilk-stripe** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_IsbLqDfWWCINKf
→ https://go/builds/bui_IsbLbpAKwhw0Tm

File: core/Symbols.cc [hunk 1]
```diff
@@ -1341,9 +1341,19 @@ u4 Symbol::hash(const GlobalState &gs) const {
...
+        vector<core::SymbolRef> membersToHash;
+        membersToHash.reserve(this->members().size());
+        for (auto e : this->members()) {
+            if (e.second.exists() && !e.second.data(gs)->ignoreInHashing(gs)) {
+                membersToHash.emplace_back(e.second);
+            }
+        }
+        fast_sort(membersToHash, [](const auto &a, const auto &b) -> bool { return a.rawId() < b.rawId(); });
+        for (auto member : membersToHash) {
+            result = mix(result, _hash(member.data(gs)->name.shortName(gs)));
```
Comment (jez) [inline core/Symbols.cc:1356]: I see that the old code did this, but why are we using short name here?

That would mean that if a name changes from being a constant name to a utf8 name (which would no longer be the same name for many purposes), the hash would remain unchanged.

Comment (jvilk-stripe) [reply core/Symbols.cc:1356]: Gooooooooooooooood question.

Comment (jvilk-stripe) [reply core/Symbols.cc:1356]: I can re-introduce the NameRef::contentHash from my other PR if that helps?

Comment (jvilk-stripe) [reply core/Symbols.cc:1356]: Is there a case where this would change and Sorbet doesn't remove or delete a symbol?

Comment (jez) [reply core/Symbols.cc:1356]: @jvilk-stripe Not quite sure, I can think about it later if you need me to.

Comment (jvilk-stripe) [reply core/Symbols.cc:1356]: I think the answer is 'no', but directionally speaking it'd be good to include a comprehensive content-based hash of the name.

**Review by jez — APPROVED** [review]
thanks!

**Pull request merged**